src/program/processor/new_order.rs

```diff
@@ -325,6 +325,58 @@ fn process_new_order<'a, 'info>(
         let mut get_clock_fn = || (clock.slot, clock.unix_timestamp as u64);
         let market_bytes = &mut market_info.try_borrow_mut_data()?[size_of::<MarketHeader>()..];
         let market = load_with_dispatch_mut(&market_info.size_params, market_bytes)?.inner;
+
+        // If the trader does not have sufficient funds to place the order, return silently without mutating the book.
+        if order_packet.fail_silently_on_insufficient_funds() {
+            let trader_index = market
+                .get_trader_index(trader.key)
+                .ok_or(PhoenixError::TraderNotFound)?;
+            let quote_lots_free = market
+                .get_trader_state_from_index(trader_index)
+                .quote_lots_free;
+            let base_lots_free = market
+                .get_trader_state_from_index(trader_index)
+                .base_lots_free;
```

Comment:
nit: can we fetch these in one go?

Reply:
alternately just put the whole thing inside the Side match?

Reply:
yeah, was a little lazy here, can make the code a bit prettier